pkg/ingester/ingester.go

```diff
@@ -459,6 +459,20 @@ func (i *Ingester) QueryStream(req *client.QueryRequest, stream client.Ingester_
 		return err
 	}
 
+	now := model.Now()
+	if from.After(now) {
+		// time-span start is in future ... regard as legal
+		level.Info(log).Log("msg", "whole timerange in future, yield empty resultset", "through", through, "from", from, "now", now)
+		return nil
+	}
+
+	maxFutureThrough := now.Add(5 * time.Minute)
+	if through.After(maxFutureThrough) {
+		// time-span end is in future ... regard as legal
+		level.Info(log).Log("msg", "adjusting end timerange from future to now", "old_through", through, "new_through", maxFutureThrough)
```

Comment:
in the message should we say `from future to now+5 minutes` just to be explicitly clear?